components/gc-subway/gc-subway.js

```diff
@@ -35,93 +27,67 @@ var $document = wb.doc,
 		// returns DOM object = proceed with init
 		// returns undefined = do not proceed with init (e.g., already initialized)
 		var elm = wb.init( event, componentName, selector ),
-			h1,
 			$elm;
 
 		if ( elm && event.currentTarget === event.target ) {
 			$elm = $( elm );
-			$h1 = $( "h1", $elm );
-			h1 = $h1.get( 0 );
 
-			// Add Subway H1 to skip links only once and if it is a sub-page
-			if ( h1 ) {
+			sectionsTitle = elm.hasAttribute( "data-sections-title" ) ? elm.getAttribute( "data-sections-title" ) : "Sections";
 
-				// Ensure the element have an ID
-				h1.id = h1.id || wb.getId();
+			// If it's the index, add "gc-subway-index" class
+			if ( !elm.querySelector( "h1" ) ) {
+				if ( !elm.classList.contains( indexClass ) ) {
+					elm.classList.add( indexClass );
+				}
+			} else {
+				$navH1 = $( "h1", $elm );
 
-				wb.addSkipLink( wb.i18n( "skip-prefix" ) + " " + h1.textContent, { href: "#" + h1.id } );
-			}
+				// Add skip link to sections list
+				if ( $navH1 ) {
+					$navH1.id = $navH1.id || wb.getId(); // Ensure the element has an ID
+					wb.addSkipLink( wb.i18n( "skip-prefix" ) + " " + $navH1.text(), { href: "#" + $navH1.id } );
+				}
 
-			// trigger resizing
-			onResize( $elm );
+				// Wrap all content until it hits either: ".pagedetails, .gc-subway-landmark-end, .gc-subway-support"
+				$elm.nextUntil( ".pagedetails, .gc-subway-landmark-end, .gc-subway-support" ).wrapAll( "<section class='" + mainClass + "'>" );
 
-			// Identify that initialization has completed
-			wb.ready( $elm, componentName );
-		}
-	},
+				$pageH1 = $( "." + mainClass + " h1" );
 
-	/**
-	 * Mutate DOM depending on breakpoint
-	 * @method onResize
-	 * @param {jQuery DOM element | jQuery Event} $elm Element targetted by this plugin, which is the nav | Resizing event
-	 */
-	onResize = function( $elm ) {
+				$pageH1.wrap( "<hgroup></hgroup>" );
+				$navH1.wrap( "<hgroup></hgroup>" );
 
-		if ( !$elm.length ) {
-			$elm = $( selector );
-		}
+				$( "<p>" + $navH1.text() + "</p>" ).insertBefore( $pageH1 );
+				$( "<p class='h3 hidden-xs visible-md visible-lg mrgn-tp-0'>" + sectionsTitle + "</p>" ).insertAfter( $navH1 );
 
-		// Ensure the page contains at least two heading level 1
-		if ( $( "main h1" ).length < 2 ) {
-			$document.off( wb.resizeEvents, onResize );
-			$elm.addClass( "no-blink p-0" );
-			return;
-		}
+				$elm.find( "a.active" ).attr( { tabindex: "0", "aria-current": "page" } );
+
+				$subwayLinks = $( selector + " a, ." + mainClass + " .gc-pagination a" );
 
-		// Desktop view, setup and mutate H1s
-		if ( $html.hasClass( views.md ) || $html.hasClass( views.lg ) ||
-			$html.hasClass( views.xl ) ) {
+				// Duplicating GC-Subway links for single-page application feel on mobile
+				$subwayLinks.each( function( i, el ) {
+					let $el = $( el ),
+						elHref = $el.attr( "href" ),
+						cloneHref = elHref.includes( "#" ) ? elHref : elHref += "#wb-cont";
```

Comment:
Should this also build a secondary link in the pagination to go to the second h1?

Reply:
@Christopher-O You are right, it should. As you see on line 64, it gets the links within the pagination. Unfortunately, I removed the class not remembering it was used in the JS. We'll fix it.